File: Day-1/twoSum.js
function twoSum(nums, target) {
    let numMap = new Map();
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i];
        if (numMap.has(complement)) {
            return [numMap.get(complement), i];
        }
        numMap.set(nums[i], i);
    }
    return [];
}

// Test
console.log(twoSum([2,7,11,15], 9)); // Output: [0,1]

File: Day-1/two_sum.py
def twoSum(nums, target):
    num_map = {} # Dictionary to store numbers and their indices
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i] # Return indices
        num_map[num] = i
    return []

# Test
print(twoSum([2,7,11,15], 9))  # Output: [0,1]